helpers/parallel_processing.py

```diff
@@ -0,0 +1,20 @@
+"""
```

Comment:
Is this really `parallel_processing.py` or a specific orchestrator for a specific test?  I feel it is the latter. 

I could be wrong, but I think you are thinking along the lines of having reusable orchestrators and then calling them in the test. This could work but it could also be very confusing for folks that are not used to this pattern. Anytime we introduce async stuff, it can get tricky when mixing and matching different methods. In this specific PR, it is ok since we have only one test.